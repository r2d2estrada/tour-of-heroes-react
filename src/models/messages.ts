export interface Message {
  id: number;
  message: string;
}
