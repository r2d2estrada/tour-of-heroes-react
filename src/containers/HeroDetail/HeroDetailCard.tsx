import React from "react";
import { Hero } from "../../models/hero";
import { useHistory } from "react-router-dom";

interface HeroDetailCardProps {
  hero: Hero;
  handleChange: any;
  handleSubmit: any;
  isUpdating: boolean;
}

const HeroDetailCard: React.FC<HeroDetailCardProps> = ({
  hero,
  handleChange,
  handleSubmit,
  isUpdating,
}) => {
  const history = useHistory();

  return (
    <div className="card">
      <form onSubmit={handleSubmit}>
        <div className="card-body">
          <div className="form-inline">
            <div className="form-group">
              <label htmlFor="hero-id">ID:</label>
              <span className="text-muted ml-3">{hero.id}</span>
            </div>
            <div className="form-group ml-3">
              <label htmlFor="hero-name">NAME:</label>
              <input
                name="name"
                type="text"
                className="form-control ml-3"
                id="hero-name"
                value={hero.name}
                onChange={handleChange}
              />
            </div>
          </div>
        </div>
        <div className="card-footer text-right">
          <button
            type="button"
            className="btn btn-light"
            onClick={() => history.goBack()}
          >
            Back
          </button>
          <button
            type="submit"
            className="btn btn-primary ml-3"
            disabled={isUpdating}
          >
            {isUpdating ? "Saving..." : "SAVE"}
          </button>
        </div>
      </form>
    </div>
  );
};

export default HeroDetailCard;
