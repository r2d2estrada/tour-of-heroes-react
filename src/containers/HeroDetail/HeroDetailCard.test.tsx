import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import HeroDetailCard from "./HeroDetailCard";
import { Hero } from "../../models/hero";

const mockHistoryPush = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useHistory: () => ({
    push: mockHistoryPush,
    goBack: mockHistoryPush,
  }),
}));

describe("HeroDetailCard", () => {
  let component: any;
  let mockHero: Hero;

  beforeAll(() => {
    mockHero = {
      id: 1,
      name: "Test",
    };

    component = (
      <MemoryRouter>
        <HeroDetailCard
          hero={mockHero}
          handleChange={jest.fn()}
          handleSubmit={jest.fn}
          isUpdating={false}
        />
      </MemoryRouter>
    );
  });

  it("should render component and match snapshot", () => {
    const { container } = render(component);
    expect(container).toMatchSnapshot();
  });

  it("should go back when clicked is pushed", () => {
    const { getAllByRole } = render(component);
    fireEvent.click(getAllByRole("button")[0]);
    expect(mockHistoryPush).toHaveBeenCalled();
  });

  it("should submit form", () => {});
});
