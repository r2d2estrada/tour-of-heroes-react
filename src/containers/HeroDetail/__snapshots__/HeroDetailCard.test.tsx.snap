// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroDetailCard should render component and match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <form>
      <div
        class="card-body"
      >
        <div
          class="form-inline"
        >
          <div
            class="form-group"
          >
            <label
              for="hero-id"
            >
              ID:
            </label>
            <span
              class="text-muted ml-3"
            >
              1
            </span>
          </div>
          <div
            class="form-group ml-3"
          >
            <label
              for="hero-name"
            >
              NAME:
            </label>
            <input
              class="form-control ml-3"
              id="hero-name"
              name="name"
              type="text"
              value="Test"
            />
          </div>
        </div>
      </div>
      <div
        class="card-footer text-right"
      >
        <button
          class="btn btn-light"
          type="button"
        >
          Back
        </button>
        <button
          class="btn btn-primary ml-3"
          type="submit"
        >
          SAVE
        </button>
      </div>
    </form>
  </div>
</div>
`;
